oioioi/mp/admin.py

```diff
@@ -36,20 +36,24 @@ def get_actions(self, request):
         return actions
 
 
-class SubmissionScoreMultiplierInline(admin.StackedInline):
+class SubmissionScoreMultiplierInline(admin.TabularInline):
     model = SubmissionScoreMultiplier
     extra = 0
     category = _("Advanced")
 
 
 class SubmissionScoreMultiplierAdminMixin(object):
-    """Adds :class:`~oioioi.mp.SubmissionScoreMultiplier` fields to an
-    admin panel.
+    """Adds :class:`~oioioi.mp.models.SubmissionScoreMultiplier` to an admin panel
+    when contest controller is MPContestController.
     """
 
     def __init__(self, *args, **kwargs):
         super(SubmissionScoreMultiplierAdminMixin, self).__init__(*args, **kwargs)
-        self.inlines = self.inlines + [SubmissionScoreMultiplierInline]
+
+    def get_inlines(self, request, obj=None):
```

Comment:
Is it safe to run this function multiple times? I am not sure django guaranties it to be executed only once. Maybe just return self.inlines with SCMInline appended.